partitionmanager/bootstrap.py

```diff
@@ -101,6 +117,91 @@ def _plan_partitions_for_time_offsets(
     return changes
 
 
+def _suffix(lines, *, indent="", mid_suffix="", final_suffix=""):
+    """ Helper that suffixes each line with either mid- or final- suffix """
+    for line, is_final in partitionmanager.tools.iter_show_end(lines):
+        if is_final:
+            yield indent + line + final_suffix
+        else:
+            yield indent + line + mid_suffix
+
+
+def _trigger_column_copies(cols):
+    """ Helper that returns lines copying each column for a trigger. """
+    for c in cols:
+        yield f"`{c}` = NEW.`{c}`"
+
+
+def _generate_sql_copy_commands(
+    existing_table, map_data, columns, new_table, alter_commands_iter
+):
+    """ Generate a series of SQL commands to start a copy of the existing_table
+    to a new_table, applying the supplied alterations before starting the
+    triggers. """
+    log = logging.getLogger(
+        f"_generate_sql_copy_commands:{existing_table.name} to {new_table.name}"
+    )
+
+    max_val_part = map_data["partitions"][-1]
+    if not isinstance(max_val_part, partitionmanager.types.MaxValuePartition):
+        log.error(f"Expected a MaxValue partition, got {max_val_part}")
+        raise Exception("Unexpected part?")
```

Comment:
nit: the exception should contain all the same information as the log statement (I kinda wish languages had a `raise_and_log` construct).